assets/js/util/sort-object-map-by-key.js

```diff
@@ -0,0 +1,37 @@
+/**
+ * sortObjectMapByKey utiltity function.
+ *
+ * Site Kit by Google, Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Sorts an object map based on numberic order/priority key.
+ *
+ * @param {Object} objectList  Object containg objects to be sorted
+ * @param {string} priorityKey The object key to use for matching.
+ *
+ * @return {Array} An Array containing objects ordered by the priority key
+ */
+export function sortObjectMapByKey( objectList, priorityKey ) {
```

Comment:
This is great, can you replace https://github.com/google/site-kit-wp/blob/ff9c5141800adbad9a1cf26ca09b291a57bd361a/assets/js/googlesitekit/widgets/datastore/widgets.js#L224 and https://github.com/google/site-kit-wp/blob/ff9c5141800adbad9a1cf26ca09b291a57bd361a/assets/js/googlesitekit/widgets/datastore/areas.js#L204 with this new function? 😄